x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticatorTests.java

```diff
@@ -320,7 +320,7 @@ public void testClockSkewIsHonored() throws Exception {
             .issuer(opConfig.getIssuer().getValue())
             .issueTime(Date.from(now().minusSeconds(200)))
             .notBeforeTime(Date.from(now().minusSeconds(200)))
-            .claim("nonce", nonce)
+            .claim("nonce", nonce.toString())
```

Comment:
This tweak is necessary since we otherwise hit a `java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "createClassLoader")` exception. 

`createClassLoader` is called by the `com.nimbusds.jose` lib (on the new version) for serializing classes that don't have a specific writer registered (e.g. `Nonce`). `String` has a specific writer and is therefore safe.

I don't think this is a concern for production code, but I'm still validating this.

Reply:
Actually, I'm not confident anymore that this is okay for prod code:

All it takes to trigger this is creating a claim set with a class like `Nonce` in a claim and call `toString` on it and we get a class loader exception:

```
JWTClaimsSet.Builder idTokenBuilder = new JWTClaimsSet.Builder().claim("nonce", new Nonce());
        // causes java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "createClassLoader")
        expectThrows(SecurityException.class, () -> idTokenBuilder.build().toString());
```